src/move.c

```diff
@@ -2725,22 +2709,12 @@ scroll_cursor_bot(int min_scroll, int set_topbot)
      */
     if (line_count >= curwin->w_height && line_count > min_scroll)
 	scroll_cursor_halfway(FALSE, TRUE);
-    else
+    else if (line_count > 0)
     {
-	// With 'smoothscroll' scroll at least the height of the cursor line,
-	// unless it would move the cursor.
-	if (curwin->w_p_wrap && curwin->w_p_sms && line_count < min_scrolled
-		&& (curwin->w_cursor.lnum < curwin->w_topline
-		    || (curwin->w_virtcol - curwin->w_skipcol >=
-					  curwin->w_width - curwin_col_off())))
-	    line_count = min_scrolled;
```

Comment:
Unused since if 'smoothscroll' and 'wrap' are set, we are scrolling by "scrolled" lines, not "line_count". The problem this was solving also seems to be prevented elsewhere, as attested by the still passing tests that were added in https://github.com/vim/vim/commit/9bab7a024393200bb2c03b3abddfda86436990a7.